src/changelog.rs

```diff
@@ -2,6 +2,31 @@ use std::io::BufWriter;
 use clog::Clog;
 use clog::fmt::MarkdownWriter;
 use std::path::PathBuf;
+use std::io::prelude::*;
+use std::fs::{self, File, OpenOptions};
+
+fn changelog_exists(path: &PathBuf) -> bool {
+    match fs::metadata(path) {
+        Ok(md) => md.is_file(),
+        Err(_) => return false
+    }
+}
+
+fn prepend_to_file(filename: PathBuf, new_changelog: &str) -> Result<(), String> {
+    let mut existing_file_content = String::new();
+    let mut f = File::open(&filename).expect("Failed to open Changelog for reading");
+    f.read_to_string(&mut existing_file_content).expect("Failed to read Changelog");
+
+    let mut file = OpenOptions::new().create(true).write(true).open(filename).expect("Failed to open Changelog for prepending new items");
```

Comment:
Those `.expect` calls will trigger panics. We are already using a `Result`, so we could simply return the Errors (use our own [error](https://github.com/semantic-rs/semantic-rs/blob/master/src/error.rs), we handle IoError already anway)